teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -183,6 +185,43 @@ export class MapsService {
     return this.findMap(clientMap.uuid)
   }
 
+  async updateMapByDiff(mapId: string, diff: IMmpClientMapDiff) {
+    if (diff.added && typeof diff.added === 'object') {
```

Comment:
this can be done easier, I think. Just a quick sketch, feel free to improve it:

```typescript
const diffAddedCallback = () => {...}
const diffRemovedCallback = () => {...}
const diffUpdatedCallback = () => {...}

const callbacks = {
'added': diffAddedCallback,
...
}

['added', 'updated', 'removed'].each((key) => {
  if(diff[key] && diff[key] === 'object') { // testing for object isn't really best practice, aren't there better alternatives?
    callbacks[key]()
  }
})

```

Reply:
Would that make a big enough difference though? You're still going through all three keys and just moving the callbacks outside of the if, but considering it's only in one method (on the server anyway) I don't see much of a benefit here.

Reply:
it is not an efficiency/performance refactoring, it wanted to give you feedback on the style of coding and try to avoid unnecessary code duplication (DRY). Currently you have a loop which is coded 3 times with basically the same content. At least for readability, please extract the content where possible into smaller ones.

Reply:
Oh yeah I understand what you were getting at, and I wasn't talking about performance either. For me it doesn't make a massive difference whether the code is in 3 if blocks or 3 call backs, since the (minimal) duplication would still be there. I don't mind changing it if it makes it more readable for you though, since you're the one that's going to have to maintain this in a few months' time :D

Reply:
I've added callbacks now, maybe with a bit of excessive type safety 😂